nullaway/src/test/java/com/uber/nullaway/tools/SerializationTestHelper.java

```diff
@@ -157,4 +159,47 @@ private List<T> readActualOutputs(Path outputPath) {
     }
     return outputs;
   }
+
+  /**
+   * Checks if given paths are equal. Under different OS environments, identical paths might have a
+   * different string representation. In windows all forward slashes are replaced with backslashes.
+   *
+   * @param expected Expected serialized path.
+   * @param found Serialized path.
+   * @return true, if paths are identical.
+   */
+  public static boolean pathsAreEqual(String expected, String found) {
+    if (found.equals(expected)) {
+      return true;
+    }
+    return found.replaceAll("\\\\", "/").equals(expected);
+  }
+
+  /**
+   * Extracts relative path from the serialized full path.
+   *
+   * @param pathInString Full serialized path.
+   * @return Relative path to "com" from the given path including starting from "com" directory.
+   */
+  public static String getRelativePathFromUnitTestTempDirectory(String pathInString) {
+    if (pathInString.equals("null")) {
+      return "null";
+    }
+    // using atomic refs to use them inside inner class below. This is not due to any concurrent
+    // modifications.
+    AtomicReference<Path> relativePath = new AtomicReference<>(Paths.get("com"));
+    AtomicReference<Boolean> relativePathStarted = new AtomicReference<>(false);
+    Path path = Paths.get(pathInString);
+    path.iterator()
+        .forEachRemaining(
+            remaining -> {
+              if (relativePathStarted.get()) {
+                relativePath.set(relativePath.get().resolve(remaining));
+              }
+              if (remaining.toString().startsWith("com")) {
```

Comment:
Wonder if `"com"` as a directory name to match on might not clash with something, it's a lot more likely to appear elsewhere in the path than "com" followed immediately by "uber".

Couldn't we use `Path.relativize(...)` given we save the temp directory we use as `root`? Or does that break for some reason?

Reply:
In order to use `Path.relativize()` we need to know the root directory where all source files exists and `CompilatIionTestHelper` does not give us the root directory. We only know the root directory where all serialized output exist. Please let me know if I missed anything. Thank you.

Reply:
Actually I just realized that we give the output path ourself to `CompilationTestHelper`. Sorry for the mistake. Should I update it in the follow up PR or in this PR ?

Reply:
Let's leave that for a follow up, as well as the change in the comment, and wait until after the internal deadline before we fix either. It's not the highest priority thing right now, since it's test code.